src/components/overlay/index.js

```diff
@@ -0,0 +1,94 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {StyleSheet, Image} from 'react-native';
+import {Colors} from '../../style';
+import {PureBaseComponent} from '../../commons';
+import View from '../view';
+
+
+const gradientImage = require('./assets/GradientOverlay.png');
+const OVERLY_TYPES = {
+  DEFAULT: 'default',
+  TOP: 'top',
+  BOTTOM: 'bottom',
+  SOLID: 'solid'
+}
+/**
+ * @description: Overlay view with types (default, top, bottom, solid)
+ * @extends: Image
+ * @extendsLink: https://facebook.github.io/react-native/docs/image
+ */
+export default class Overlay extends PureBaseComponent {
+  static displayName = 'Overlay';
+
+  static propTypes = {
+    /** 
+    * The type of overlay to set on top of the image 
+    */
+    type: PropTypes.oneOfType([PropTypes.oneOf(Object.keys(OVERLY_TYPES)), PropTypes.string])
```

Comment:
This is not correct, it should be 
`type: PropTypes.oneOf(_.values(OVERLY_TYPES))`